tpp-rt/PerfRunnerUtils.cpp

```diff
@@ -0,0 +1,89 @@
+//===- CRunnerUtils.cpp - Utils for MLIR execution ------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Utilities to measure performance (timers, statistics, etc).
+//
+//===----------------------------------------------------------------------===//
+
+#include "PerfRunnerUtils.h"
+#include <cstdlib>
+#include <time.h>
+
+namespace {
+/// Local class to keep the results and stats of each measurement
+class PerfResults {
+  double mean = 0.0;
+  double stdev = 0.0;
+  std::vector<double> timings;
+
+  void stats() {
+    auto size = timings.size();
+    // Mean
+    double sum = 0.0;
+    for (size_t i = 0; i < size; i++)
+      sum += timings[i];
+    mean = sum / size;
+    // Stdev
+    sum = 0.0;
+    for (size_t i = 0; i < size; i++) {
+      double delta = timings[i] - mean;
+      sum += delta * delta;
+    }
+    stdev = sqrt(sum / size);
+  }
+
+public:
+  void accumulate(double val) {
+    timings.push_back(val);
+  }
+
+  double getMean() {
+    if (mean == 0.0)
+      stats();
+    return mean;
+  }
+
+  double getStdev() {
+    if (stdev == 0.0)
+      stats();
+    return stdev;
```

Comment:
I want to avoid calling it twice. I'm hoping we can find a better way to return two values in the IR (like a vector), but I left it for later.

Reply:
My primary concern here was that any new data coming from `accumulate` calls after getting the mean/stdev are ignored. But alright let's keep it simple for now.

Reply:
Ah, I see, this is obviously simplified, and there should be no new calls to accumulate after stats. Perhaps we could add an assert to make sure the API is clear.

Reply:
Turns out the `chrono` call is always zero in the cluster, but using `std::chrono` works, so I'll update that. This will change the API, which is good that we do sooner.